src/test/OperationsBase.test.ts

```diff
@@ -0,0 +1,127 @@
+  const operationsBase = new OperationsBase();
+
+  afterEach(() => {
+    sinon.restore();
+  })
+
+  it("isSimpleIdentifier", () => {
+    expect(operationsBase.isSimpleIdentifier("")).to.be.eq(false);
+    expect(operationsBase.isSimpleIdentifier(" Test")).to.be.eq(false);
+    expect(operationsBase.isSimpleIdentifier("Test 1")).to.be.eq(false);
+    expect(operationsBase.isSimpleIdentifier("Test!")).to.be.eq(false);
+    expect(operationsBase.isSimpleIdentifier("!Test")).to.be.eq(false);
+    expect(operationsBase.isSimpleIdentifier("0Test")).to.be.eq(false);
+    expect(operationsBase.isSimpleIdentifier(
+      "TestTestTestTestTestTestTestTestTestTestTestTestTestTestTestTestTestTestTestTestTestTestTestTestTestTestTestTestTestTestTestTestTestTestTestTestTestTestTestTest"
+      )).to.be.eq(false);
+    expect(operationsBase.isSimpleIdentifier("Test")).to.be.eq(true);
+    expect(operationsBase.isSimpleIdentifier("_Test")).to.be.eq(true);
+    expect(operationsBase.isSimpleIdentifier("Test0")).to.be.eq(true);
+    expect(operationsBase.isSimpleIdentifier("Test_name")).to.be.eq(true);
+    expect(operationsBase.isSimpleIdentifier("_")).to.be.eq(true);
+  });
+
+  it("isNullOrWhitespace", () => {
+    expect(operationsBase.isNullOrWhitespace("")).to.be.eq(true);
+    expect(operationsBase.isNullOrWhitespace("       ")).to.be.eq(true);
+    expect(operationsBase.isNullOrWhitespace("Test")).to.be.eq(false);
+    expect(operationsBase.isNullOrWhitespace("  Test  ")).to.be.eq(false);
+  });
+
+  it("isValid", () => {
+    const stub = sinon.stub(OperationsBase.prototype, "isNullOrWhitespace");
+    stub.returns(false);
+    stub.withArgs("").returns(true);
+
+    const prop: ECProperty = {
+      ecSchemaName: "Name",
+      ecClassName: "Class",
+      ecPropertyName: "Property",
+      ecPropertyType: DataType.Integer
+    };
+    expect(operationsBase.isValidECProperty(prop)).to.be.eq(true);
+
+    prop.ecClassName = "";
+    expect(operationsBase.isValidECProperty(prop)).to.be.eq(false);
+
+    prop.ecClassName = "Class";
+    prop.ecPropertyName = "";
+    expect(operationsBase.isValidECProperty(prop)).to.be.eq(false);
+
+    prop.ecPropertyName = "Property";
+    prop.ecSchemaName = "";
+    expect(operationsBase.isValidECProperty(prop)).to.be.eq(false);
+
+    prop.ecSchemaName = "Name";
+    prop.ecPropertyType = DataType.Undefined;
+    expect(operationsBase.isValidECProperty(prop)).to.be.eq(false);
+  });
+
+  it("fetch", async () => {
+    const stub = sinon.stub(operationsBase, <any>"fetch");  // eslint-disable-line @typescript-eslint/no-explicit-any
+    let myOptions = { status: 200, statusText: "Test" };
+    const body = {
+      "Test": "test"
+    }
+    let response: Response = new Response(JSON.stringify(body), myOptions);
+    stub.resolves(response);
+    let realResponse = await operationsBase.fetchJSON("-", {});
+    expect(realResponse).to.not.be.undefined;
+
+    myOptions = { status: 204, statusText: "Test" };
```

Comment:
It's a bit weird that you picked the 204 status (which translates to No Content) for these tests where you provide a response body. I think 200 or 201 status codes would be a better fit here.

Reply:
In the fetchJSON function the return statement is ``` return response.status === 204 ? response : response.json() ```.
The 204 response status is chosen to cover a branch.